x-pack/plugins/enterprise_search/public/applications/app_search/components/meta_engine_creation/constants.tsx

```diff
@@ -0,0 +1,119 @@
+  {
+    defaultMessage: 'Create meta engine',
+  }
+);
+
+export const META_ENGINE_CREATION_FORM_META_ENGINE_DESCRIPTION = i18n.translate(
+  'xpack.enterpriseSearch.appSearch.metaEngineCreation.form.metaEngineDescription',
+  {
+    defaultMessage:
+      'Meta engines allow you to combine multiple engines into one, searchable engine.',
+  }
+);
+
+export const META_ENGINE_CREATION_FORM_DOCUMENTATION_LINK = i18n.translate(
+  'xpack.enterpriseSearch.appSearch.metaEngineCreation.form.documentationLink',
+  {
+    defaultMessage: 'Read the documentation',
+  }
+);
+
+export const META_ENGINE_CREATION_FORM_DOCUMENTATION_DESCRIPTION = (
+  <FormattedMessage
+    id="xpack.enterpriseSearch.appSearch.metaEngineCreation.form.documentationDescription"
+    defaultMessage="{documentationLink} for information about how to get started."
+    values={{
+      documentationLink: (
+        <a
+          href="https://www.elastic.co/guide/en/app-search/current/meta-engines-guide.html"
+          target="_blank"
+        >
+          {META_ENGINE_CREATION_FORM_DOCUMENTATION_LINK}
+        </a>
+      ),
+    }}
+  />
+);
+
+export const META_ENGINE_CREATION_FORM_ENGINE_NAME_LABEL = i18n.translate(
+  'xpack.enterpriseSearch.appSearch.metaEngineCreation.form.engineName.label',
+  {
+    defaultMessage: 'Meta engine name',
+  }
+);
+
+export const ALLOWED_CHARS_NOTE = i18n.translate(
+  'xpack.enterpriseSearch.appSearch.metaEngineCreation.form.engineName.allowedCharactersHelpText',
+  {
+    defaultMessage: 'Meta engine names can only contain lowercase letters, numbers, and hyphens',
+  }
+);
+
+export const SANITIZED_NAME_NOTE = i18n.translate(
+  'xpack.enterpriseSearch.appSearch.metaEngineCreation.form.engineName.sanitizedNameHelpText',
+  {
+    defaultMessage: 'Your meta engine will be named',
+  }
+);
+
+export const META_ENGINE_CREATION_FORM_ENGINE_NAME_PLACEHOLDER = i18n.translate(
+  'xpack.enterpriseSearch.appSearch.metaEngineCreation.form.engineName.placeholder',
+  {
+    defaultMessage: 'i.e., my-meta-engine',
+  }
+);
+
+export const META_ENGINE_CREATION_FORM_ENGINE_SOURCE_ENGINES_LABEL = i18n.translate(
+  'xpack.enterpriseSearch.appSearch.metaEngineCreation.form.sourceEngines.label',
+  {
+    defaultMessage: 'Add source engines to this meta engine',
+  }
+);
+
+export const META_ENGINE_CREATION_FORM_MAX_SOURCE_ENGINES_WARNING_TITLE = (
+  maxEnginesPerMetaEngine: number
+) =>
+  i18n.translate(
+    'xpack.enterpriseSearch.appSearch.metaEngineCreation.form.sourceEngines.maxSourceEnginesWarningTitle',
+    {
+      defaultMessage: 'Meta engines have a limit of {maxEnginesPerMetaEngine} source engines',
+      values: { maxEnginesPerMetaEngine },
+    }
+  );
+
+export const META_ENGINE_CREATION_SUCCESS_MESSAGE = i18n.translate(
+  'xpack.enterpriseSearch.appSearch.metaEngineCreation.successMessage',
+  {
+    defaultMessage: 'Successfully created meta engine.',
+  }
+);
```

Comment:
After reading this, I'm starting to become more of a fan of putting all our i18n into `constants.tsx` files rather than inline - it's nice in that it focuses i18n IDs as well as makes it easier to skim a single file for grammar/sentence casing it. @JasonStoltz any thoughts here?

Reply:
Maybe a `i18n.ts`/`i18n.tsx` file?  They're still "constants" but I think if we intend to keep all of these out of the component `.tsx` files I'd rather give them a unique file (and use constants for stuff like `DEFAULT_META`)